src/org/labkey/test/tests/DomainDesignerTest.java

```diff
@@ -1498,6 +1507,172 @@ public void addUpdateRemoveConditionalFormat() throws Exception
         assertEquals(true, editedAquamap.get("bold"));
     }
 
+
+    @Test
+    public void testExportFields() throws Exception
+    {
+        String listName = "exportFieldsTestList";
+
+        FieldDefinition.LookupInfo lookupInfo = new FieldDefinition.LookupInfo(getProjectName(), "lists", listName);
+        ConditionalFormatFilter eqFive = new ConditionalFormatFilter("5", Filter.Operator.EQUAL);
+        FieldDefinition.RangeValidator soonValidator = new FieldDefinition.RangeValidator("soonValidator", "between now and 2 days from now", "aaaa",
+                FieldDefinition.RangeType.GTE, "2020-11-04",
+                FieldDefinition.RangeType.LTE, "2020-11-06");
+        TestDataGenerator dgen = new TestDataGenerator(lookupInfo)
+                .withColumns(List.of(
+                        new FieldDefinition("textField", FieldDefinition.ColumnType.String)
+                            .setDescription("is texty").setLabel("text field"),
+                        new FieldDefinition("booleanField", FieldDefinition.ColumnType.Boolean)
+                            .setLabel("Boolean Field")
+                            .setRequired(true)
+                            .setURL("fake/list-def.url"),
+                        new FieldDefinition("attachmentField", FieldDefinition.ColumnType.Attachment)
+                            .setHidden(true),
+                        new FieldDefinition("decimalField", FieldDefinition.ColumnType.Decimal).
+                            setScale(1234).setFormat("0.####"),
+                        new FieldDefinition("intField", FieldDefinition.ColumnType.Integer)
+                            .setMvEnabled(true)
+                            .setConditionalFormats(List.of(new ConditionalFormat(List.of(eqFive), true, true, false))),
+                        new FieldDefinition("dateTimeField", FieldDefinition.ColumnType.DateAndTime)
+                            .setValidators(List.of(soonValidator))));
+        dgen.createList(createDefaultConnection(), "Key");
+        DomainDesignerPage domainDesignerPage = DomainDesignerPage.beginAt(this, getProjectName(), "lists", listName);
+        DomainFormPanel domainFormPanel = domainDesignerPage.fieldsPanel();
+
+        File exportFile = domainFormPanel.clickExportFields();
+        List<PropertyDescriptor> createdFields = getFieldsFromExportFile(exportFile);
+
+        PropertyDescriptor stringField = createdFields.stream().filter(a-> a.getName().equals("textField"))
+                .findFirst().orElse(null);
```

Comment:
This filter to findFirst works, but we should expect a specific order to the fields that are exported and read in from the JSON file. So would it make sense to get the field at a specific index (zero here) since we are using an ArrayList in the getFieldsFromExportFile? This would also then provide a check that the exported fields are in the correct/expected ordering.

Reply:
I hadn't been enforcing order because field order here is initially established by the CreateDomainCommand, I don't know that the order isn't shuffled along the line while serializing.
I have added code to ensure that field order in the export file matches field order in the UI, also that it is maintained over a round-trip